packages/sycamore-core/src/generic_node.rs

```diff
@@ -127,6 +127,17 @@ pub trait GenericNode: fmt::Debug + Clone + PartialEq + Eq + Hash + 'static {
     /// variables after the scope is disposed.
     fn event<'a, F: FnMut(Self::EventType) + 'a>(&self, cx: Scope<'a>, name: &str, handler: F);
 
+    /// Add a event handler to the event `name`.
+    /// The event should be removed once the scope is disposed, as to prevent accessing scope
+    /// variables after the scope is disposed.
+    /// This is the boxed version in case the handler is already boxed
+    fn boxed_event<'a>(
+        &self,
+        cx: Scope<'a>,
+        name: &str,
+        handler: Box<dyn FnMut(Self::EventType) + 'a>,
+    );
+
```

Comment:
I think adding this extra method is a bit unnecessary since the performance overhead of an extra box is negligible anyways.

Reply:
You're right, I have a tendency to over-optimize when I write code. I removed the method.